docs/proposals/28-autonomous-shoot-clusters.md

```diff
@@ -0,0 +1,495 @@
+---
+title: Autonomous Shoot Clusters
+gep-number: 28
+creation-date: 2024-09-17
+status: implementable
+authors:
+- "@rfranzke"
+- "@ScheererJ"
+- "@vlerenc"
+---
+
+# GEP-28: Autonomous Shoot Clusters
+
+## Table of Contents
+
+- [GEP-28: Autonomous Shoot Clusters](#gep-28-autonomous-shoot-clusters)
+  - [Table of Contents](#table-of-contents)
+  - [Summary](#summary)
+  - [Motivation](#motivation)
+    - [Goals](#goals)
+    - [Non-Goals](#non-goals)
+  - [Proposal](#proposal)
+    - [Prerequisites](#prerequisites)
+    - [Commands](#commands)
+      - [`gardenadm discover`](#gardenadm-discover)
+      - [`gardenadm init`](#gardenadm-init)
+        - [Problems \& Solutions](#problems--solutions)
+          - [Faking a Kubernetes API](#faking-a-kubernetes-api)
+          - [Host Network vs. Pod Network](#host-network-vs-pod-network)
+          - [Dealing with Webhooks for Control Plane Components](#dealing-with-webhooks-for-control-plane-components)
+          - [`etcd` Management via `etcd-druid`](#etcd-management-via-etcd-druid)
+      - [`gardenadm token`](#gardenadm-token)
+      - [`gardenadm join`](#gardenadm-join)
+      - [`gardenadm connect`](#gardenadm-connect)
+      - [`gardenadm version`](#gardenadm-version)
+    - [`Shoot` API](#shoot-api)
+    - [Scenarios](#scenarios)
+      - [High Touch](#high-touch)
+      - [Medium Touch](#medium-touch)
+        - [`gardenadm bootstrap`](#gardenadm-bootstrap)
+    - [Disaster Recovery](#disaster-recovery)
+      - [Single-Node Failure](#single-node-failure)
+      - [Multi-Node Failure](#multi-node-failure)
+      - [Data Center Failure](#data-center-failure)
+  - [Future Work](#future-work)
+    - [Autonomous Shoot Clusters for End-Users](#autonomous-shoot-clusters-for-end-users)
+    - [Network Connectivity to Garden Cluster](#network-connectivity-to-garden-cluster)
+    - [Changes to Shoot Components](#changes-to-shoot-components)
+  - [Alternatives](#alternatives)
+    - [Using Existing Tools](#using-existing-tools)
+    - [Using Existing Services](#using-existing-services)
+    - [Using Existing Gardener](#using-existing-gardener)
+
+## Summary
+
+Gardener ships with functionality that would allow the creation of "autonomous" shoot clusters.
+Such "autonomous" clusters would be regular shoot clusters with the main difference that their control plane runs on dedicated nodes within the cluster itself (instead of a seed cluster which typically resides in another infrastructure account and network domain).
+This deployment model is known as the typical/standard Kubernetes deployment model, while Gardener's regular shoot clusters today follow the "Kubeception" deployment model.
+
+![Overview](assets/28-overview.png)
```

Comment:
From what I get reading this GEP it seems a big  portion of the value of autonomous clusters comes from gardenlet (initial clusters with operator and non-initial clusters) as it drives reconciliation that provides cluster management and upgrades via the `Shoot` API . I think it makes sense gardenlet to be part of the image for an autonomous shoot cluster.

Reply:
I agree with @ScheererJ. As for regular shoots, `gardenlet` is responsible for reconciliation, however, here we only focus on the cluster architecture. The important part is that, unlike it's done today for regular shoots, the control plane runs in the same network domain like the worker nodes. What component is managing this is not the main focus.